src/node.cc

```diff
@@ -448,20 +453,47 @@ void LoadEnvironment(Environment* env) {
 
 
 #ifdef __POSIX__
+
+static std::atomic<struct sigaction> previous_sigsegv_action;
+
+void TrapWebAssemblyOrContinue(int signo, siginfo_t* info, void* ucontext) {
+  if (!v8::TryHandleWebAssemblyTrapPosix(signo, info, ucontext)) {
+    auto prev = previous_sigsegv_action.load();
```

Comment:
Sorry, I should have mentioned in my previous review that just any `std::atomic<T>` might not be sufficient because it's allowed to use mutexes internally and that's not safe inside a signal handler (can deadlock).

IOW, `std::atomic<T>::is_lock_free()` _must_ be `true` and you don't get that guarantee with compound types like `struct sigaction`.

However, if you store just the function pointer, it should be okay (but please still `CHECK(prev_sigsegv_action.is_lock_free())` :-))

Reply:
signal handlers horrify me :(